tests/test91.c

```diff
@@ -0,0 +1,51 @@
+#include <stdio.h>
+#include "utlist.h"
+
+typedef struct el {
+  int id, score;
+  struct el *next, *prev;
+} el;
+
+
+static int order_desc(el *a, el *b) {
+  if (a->score < b->score) return 1;
+  return -1;
```

Comment:
FYI: I think it's important to test the "elements with cmp==0" case, so I adjusted this test in my branch.